ui-core/src/components/inputs/InputWithSuggestions.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {
+  useState,
+  useRef,
+  useEffect,
+  ChangeEventHandler,
+  KeyboardEventHandler,
+  ReactNode,
+  FocusEventHandler,
+} from 'react';
+import cx from 'classnames';
+import Input, { InputProps } from './Input';
+
+interface InputWithSuggestionsProps<T> extends Omit<InputProps, 'onChange' | 'value'> {
+  suggestions: Array<T>;
+  onChange: (value: T) => void;
+  getSuggestionLabel: (option: T) => string;
+  customLabel?: ReactNode;
+  numberOfSuggestionsToShow?: number;
+  exactSearch?: boolean;
+}
+
+const InputWithSuggestions = <T,>({
+  suggestions,
+  onChange,
+  getSuggestionLabel,
+  customLabel,
+  numberOfSuggestionsToShow = 5,
+  exactSearch = false,
+  ...inputProps
+}: InputWithSuggestionsProps<T>) => {
+  const [filteredSuggestions, setFilteredSuggestions] = useState<T[]>([]);
+  const [activeSuggestion, setActiveSuggestion] = useState(-1);
+  const [showSuggestions, setShowSuggestions] = useState(false);
```

Comment:
Is this state really necessary? It seems like `showSuggestions` is correlated with `filteredSuggestions`.

Reply:
Not really it i s used for the blur

Reply:
Fixed